### PR TITLE
Decompile most of code_b72b0_len_15ed0.c

### PR DESCRIPTION
**Pull request by pixel-stuck**

### CODE REVIEW

File: src/code_b72b0_len_15ed0.c [hunk 1]
```diff
@@ -9,17 +13,198 @@ extern s32 D_80154388; // entity fog alpha
...
+    {
+        EntityModel** entityModel = (*gCurrentEntityModelList + i);
+        s32* entityModelCount = &gEntityModelCount;
+        (*gCurrentEntityModelList)[i] = newEntityModel = heap_malloc(sizeof(EntityModel));
+        (*entityModelCount)++;
+    }
+    ASSERT(newEntityModel != NULL);
+
+    newEntityModel->flags = 0x17;
+    newEntityModel->renderMode = (u8)1;
```
Comment (ethteck) [inline src/code_b72b0_len_15ed0.c:81]: this cast should be unnecessary

File: src/code_b72b0_len_15ed0.c [hunk 2]
```diff
@@ -9,17 +13,198 @@ extern s32 D_80154388; // entity fog alpha
...
+}
+
+void update_entity_rendercmd(s32 idx) {
+    EntityModel* entityModel;
+    void* temp_v0_2;
+
+    if (!gGameStatusPtr->isBattle || (idx & 0x800)) {
+        idx &= ~0x800;
+        entityModel = (*gCurrentEntityModelList)[idx];
+        if (entityModel != NULL) {
```
Comment (ethteck) [inline src/code_b72b0_len_15ed0.c:149]: I think these conditions should be anded together

File: src/code_b72b0_len_15ed0.c [hunk 3]
```diff
@@ -41,25 +226,101 @@ INCLUDE_ASM(s32, "code_b72b0_len_15ed0", draw_entity_model_extra2);
...
+        free_entity_model_by_index(i);
+    }
+}
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", clear_entity_model_flags);
+void set_entity_model_flags(s32 idx, s32 newFlags) {
+    EntityModel* entityModel = (*gCurrentEntityModelList)[idx & ~0x800];
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", func_80122F64);
+    if (entityModel != NULL) {
```
Comment (ethteck) [inline src/code_b72b0_len_15ed0.c:289]: these can be anded

File: src/code_b72b0_len_15ed0.c [hunk 4]
```diff
@@ -41,25 +226,101 @@ INCLUDE_ASM(s32, "code_b72b0_len_15ed0", draw_entity_model_extra2);
...
+        if (entityModel->flags) {
+            entityModel->flags |= newFlags;
+        }
+    }
+}
+
+void clear_entity_model_flags(s32 idx, s32 newFlags) {
+    EntityModel* entityModel = (*gCurrentEntityModelList)[idx & ~0x800];
+
+    if (entityModel != NULL) {
```
Comment (ethteck) [inline src/code_b72b0_len_15ed0.c:299]: these too

File: src/code_b72b0_len_15ed0.c [hunk 5]
```diff
@@ -110,22 +383,168 @@ void init_dynamic_entity_list(void) {
...
+    }
+}
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", update_dynamic_entities);
+void render_dynamic_entities(void) {
+    s32 i;
+
+    for (i = 0; i < MAX_DYNAMIC_ENTITIES; i++) {
+        DynamicEntity* entity = (*gCurrentDynamicEntityListPtr)[i];
+        if (entity != NULL) {
```
Comment (ethteck) [inline src/code_b72b0_len_15ed0.c:496]: these can be anded too

File: src/code_b72b0_len_15ed0.c [hunk 6]
```diff
@@ -110,22 +383,168 @@ void init_dynamic_entity_list(void) {
...
+    }
+}
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", render_dynamic_entities);
+void func_801234E0(void) {
+    s32 i;
+
+    for (i = 0; i < MAX_DYNAMIC_ENTITIES; i++) {
+        DynamicEntity* entity = (*gCurrentDynamicEntityListPtr)[i];
+        if (entity != NULL) {
```
Comment (ethteck) [inline src/code_b72b0_len_15ed0.c:511]: here2

File: src/code_b72b0_len_15ed0.c [hunk 7]
```diff
@@ -110,22 +383,168 @@ void init_dynamic_entity_list(void) {
...
+    }
+}
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", func_801234E0);
+void func_80123550(void) {
+    s32 i;
+
+    for (i = 0; i < MAX_DYNAMIC_ENTITIES; i++) {
+        DynamicEntity* entity = (*gCurrentDynamicEntityListPtr)[i];
+        if (entity != NULL) {
```
Comment (ethteck) [inline src/code_b72b0_len_15ed0.c:526]: here3

File: src/code_b72b0_len_15ed0.c [hunk 8]
```diff
@@ -9,17 +13,198 @@ extern s32 D_80154388; // entity fog alpha
...
+        gCurrentEntityModelList = &gWorldEntityModelList;
+    } else {
+        gCurrentEntityModelList = &gBattleEntityModelList;
+    }
+
+    for (i = 0; i < MAX_ENTITY_MODELS; i++) {
+        (*gCurrentEntityModelList)[i] = NULL;
+    }
+
+    D_8015437C = 0xA;
```
Comment (ethteck) [inline src/code_b72b0_len_15ed0.c:31]: We've been using decimal for non-flag values, so I think all these numbres here and in init_entity_models should be changed

**Review by ethteck — CHANGES_REQUESTED** [review]
Nice! mostly just anding stuff together and hex to dec, but other than that, lgtm!

**Review by ethteck — APPROVED** [review]

File: src/code_b72b0_len_15ed0.c [hunk 9]
```diff
@@ -110,22 +372,162 @@ void init_dynamic_entity_list(void) {
...
+}
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", func_80123550);
+void render_dynamic_entities(void) {
+    s32 i;
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", func_801235C0);
+    for (i = 0; i < MAX_DYNAMIC_ENTITIES; i++) {
+        DynamicEntity* entity = (*gCurrentDynamicEntityListPtr)[i];
+        if ((entity != NULL) && !(entity->flags & 2)) {
```
Comment (nanaian) [inline src/code_b72b0_len_15ed0.c:485]: Can you remove the parens around `(entity != NULL)` here and the below function? I'd rather not have extraneous parentheses

**Review by nanaian — CHANGES_REQUESTED** [review]

**Review by nanaian — APPROVED** [review]

**Pull request merged**